packages/runtime/src/widgets/SignInWithGoogle.tsx

```diff
@@ -0,0 +1,72 @@
+import { useCallback, useMemo } from "react";
+import { GoogleLogin } from "@react-oauth/google";
+import { jwtDecode } from "jwt-decode";
+import { useEnsembleUser } from "@ensembleui/react-framework";
+import type { EnsembleAction } from "@ensembleui/react-framework";
+import { WidgetRegistry } from "../registry";
+import { useEnsembleAction } from "../runtime/hooks/useEnsembleAction";
+
+export type SignInWithGoogleProps = {
+  type?: "standard" | "icon";
+  theme?: "outline" | "filled_blue" | "filled_black";
+  size?: "large" | "small" | "medium";
+  text?: "signin_with" | "signup_with";
+  shape?: "rectangular" | "pill" | "circle" | "square";
+  onSignedIn?: EnsembleAction;
+  onError?: EnsembleAction;
+};
+
+export const SignInWithGoogle: React.FC<SignInWithGoogleProps> = (props) => {
+  const [user, setUser] = useEnsembleUser();
+  const onSignInAction = useEnsembleAction(props.onSignedIn);
+  const onErrorAction = useEnsembleAction(props.onError);
```

Comment:
Need to call useRegisterBindings with the props and use the `values` object returned for the inner props. See other widgets for an example